### PR TITLE
Simple develop

### PR DESCRIPTION
**Pull request by Valc**
Hi, :)
This is a more simple and less complex to use way than allow handle changes in language files from Github.

The main config is handled from Com_localise >> "Options" tab.

To make it run is required set "Allow develop" to "Yes"

Going to "Localise"  >> "Translations" view and after choose a client and tag the program will start getting the files from Github for that selected client (try getting all clients files at same time normally crash).

develop1

One time is ready we can see what files have changes at Github.

develop2

After edit here is the output for new keys
develop3

And here for text changes:

develop4

Editing the "en-GB" tag we can handle text changes saving the reference in develop there (and no more notice about it):

develop5

Note that the new keys in develop, one time the language file is saved, will not be showed as new again.

Also note that i am using new images from Github and maybe is required choose another ones.
It is the draft.

Regards, Valc

### CODE REVIEW

**Comment by infograf768** [comment]
Will test asap!

**Comment by infograf768** [comment]
Great work and much easier than previous proposed solution! :+1: 

Important: it is necessary to use such a patched com_localise on an instance of Joomla corresponding to the last stable release. (I first tested on staging and, evidently, did not get any changed strings...)

=> Would be great if we had a similar way of knowing text changes as you do in TTM diff tool by choosing in Options with which stable source version we should compare. If possible, then we can use it on master/staging and test results of changes almost on the fly.

Default target github version should be master rather than staging.

Improvements to make it easier:
1. Filtering by "text changes" when editing an ini (btw I would rather use "Modified values" instead of text changes)
2. Displaying a supplementary way to the tooltip in the translations manager to know if there are "changed" strings values in the files. Another icon maybe in the Translated column, below the percent?

Important!
Is the "Github branch" parameter field in Options=>tab "Develop details" limited to en-GB in core?
In that case (which is fine imho) we could rather use the term "en-GB development reference language" instead of "develop".
Also, and this is important, as it will work ONLY if the "Reference language" is set to en-GB in the Options=>Global Configuration, we have to state this in a way or another in both tabs in an obvious way (I mean not only in a tooltip).

**Comment by Valc** [comment]
Hi, @infograf768:

Thanks for your feedback and comments.

All the suggestions over this new draft can to be done without problem.

The fast to solve:
- Change "staging" to "master", or create a fixed list of available branches to choose.
- Filtering by text changes.
- Use "Modified values" instead of "Text changes"
- Add an icon for "Modified values"
- Use the term "en-GB development reference language" due the stuff in this way is limited to en-GB in core.
- Send a warning when "Reference tag" is distinct to "en-GB" (from getGithubfiles function now is returning "false" when the case is detected, but without send the notice), and maybe i forgot add the same control from the models or views files that i have modified.

The other suggestions have more task and maybe will be limited:
The diff between TTM Diff tool is that it can get packages from Github or not, and in the actual case we are limited to get language files that are stored at Github with him SHA id and related tag to mount the list of "Knowed Joomla releases", so here the list will be shortest.

But i think is good enough with the available ones and a great idea than will allow us to know the diff from those to the new ones.

Is more time and code to do it, but not more complex.

So, i will start with the fast to do and then i will follow with the other one. 

Regards, Valc

**Comment by infograf768** [comment]
Thanks.

Concerning 2. took time to find the github icon as it is a binary and I could not get it by .diff
I have used the "icon-16-notice-note" instead of the github one as I did not like it :)
Changed the translations view accordingly:
`<br /><span class="icon-16-notice-note hasTooltip" title="<?php echo JText::_('COM_LOCALISE_TOOLTIP_GITHUB'); ?>"></span>`

added the css in localise.css:

```
span.icon-16-notice-note {
    background: url("../../../media/com_localise/images/icon-16-notice-note.png");
    background-repeat: no-repeat;
    height: 16px !important;
    vertical-align: text-top;
    width: 16px !important;
}
```

and copied the file in the localise images folder
I get:
screen shot 2015-05-24 at 16 38 58

**Comment by Valc** [comment]
I also like more your chooses image :D

Now we have done the fast to do:

Filtering by text changes:

develop6

Sending message when reference language is distinct to en-GB

develop7

And adding more basic info about how is running
develop8

Your changes that now also are showing by tooltip the amount of present new strings or text changes for fast review.
develop10

You have write access to my repository, so feel you free to add improvement or details when required.

Regards, Valc.

**Comment by infograf768** [comment]
I corrected some strings.
Let's finish up before introducing JPlural.

**Comment by Valc** [comment]
Thanks JM, your changes are also updated.

Now is ready the first part, but not finished the task (i have only the weekend to make it).

This means that seems it is working fine with the "Translations" and "Translation" views (get other Joomla releases as reference, display the news against last in develop, edit and save files...)

But is pending to revise the "Packages" or "Languages" views, if affected by the new features code.

From Localise >> Options >> Global configuration tab is added the "Customised en-GB files" field than allow choose from the list the Joomla version to use. The default set is the installed one.

The list is not complete, but can to be updated from 'getCustomisedtarget' function at localise helper and we can add any other Jooomla release stored at Github.

Need we add 2.5.x series?

customref1

One time we are using a customised reference, from 'en-GB' tag we can know what release is used.

customref2

Note that now if we are using the customised option, the alternative global folders in "media" path only have that set of customised core files to display (and other ones coming from non core extensions maybe installed within normal global paths are not added in "Translations" view).

The language files installed within "local" or extension folders are not affected by this one and also will be displayed, if present. 

I think is good leave it working in this way for users than have translated an older Joomla version and they only wanna know the news in core to leave equal the xx-XX  with the last one in reference. 

But i also can force it to add the non core files presents in main reference to the customised reference if you like it more.

Note that now the feature 'customised reference' also can works with 'allow develop' configured to 'No'. So, in this case i don't know how many useful can to be it.  If not useful, we can force to use  'Allow develop' configured to 'Yes' when 'Customised reference' is distinct to 'Local instance'. Now is allowed also with 'No' due i am not sure about if it is useful or not.

And also note that the program is storing customised references by each Joomla release to choose from the list, due use only one path for all them make it run very slow for each time than we make a change in 'Customised en-GB files' list.

The pending to do as second part the next weekend:
- Revise "Download package" or "fields" files when they are mounting the list of files (or any other file that is calling 'LOCALISEPATH').
- Revise client installation.
- Add Jtext.
- Details.

Please, let me know any stuff before i start second part :)

Regards, Valc.

**Comment by infograf768** [comment]
thanks! 
not sure I understand all you wrote.
will test asap

**Comment by Valc** [comment]
Yep, i have modified a bit the previous comment.

Regards, Valc.

**Comment by infograf768** [comment]
Sorry, Valc. I can't get this to work.

I installed it this time on a test site updated to staging/master.
I installed a fr-FR 3.4.1 pack.

What shall I do to get the "Modified values"?

**Comment by Valc** [comment]
To get the modified values 'Allow develop' always should be configured to 'Yes'. This field always is pointing to the newest language files versions.

From 'Customised en-GB files' You can use the installed language files by default or choose other Joomla versions. This field always is pointing to older or equal language files versions than in 'staging' or 'master' trunks. 

On an master installation If you choose at 'Customised en-GB files' the Joomla 3.4.1 version when 'Allow develop' is configured to 'Yes' and 'Joomla Github branch' is 'master' you can  know the diff from 3.4.1 vs the last in master trunk.

So 'Customised en-GB files' is working as the files that we have installed and 'Joomla Github branch' as the newest language files version against what to do the test.

If source version and target version is equal there is no diff to show.

**Comment by infograf768** [comment]
Except for the version vs which the comparison is done, it works.
I.e. I do not get
screen shot 2015-06-03 at 08 09 09

but
screen shot 2015-06-03 at 08 15 25

Looks like your screenshots compare both en-GB instances and not the language we work on compared with an en-GB instance.
In that case, I get

screen shot 2015-06-03 at 08 18 53

And there is no filter to get the modified values or untranslated (new) strings when editing such an en-GB file.

I suggest to aim at the following action:
When we compare and solve a Modified value (by changing the value or NOT, as it is often only an en-GB grammar stuff...), we should have a way to mark that string as translated (or the file complete again) in order to not have to edit again all en-GB modified values in the en-GB files to get rid of the icon. That step is really a heavy load.
Also, we should have a new filter in the Translations view to display only files with untranslated  OR/AND modified strings in the Manager.

**Comment by Valc** [comment]
"Except for the version vs which the comparison is done, it works."
It is odd, i think is not required do a  Localise >> Packages >> "Purge database" action to start showing the customised en-GB path, but maybe you can test if with purge database it is showed when en-GB tag is showed.

If not, no problem due i have founded another improved way to make it run, and your commented issue will be not possible to happens with the new mode to make it run.

With the new mode, this than now is showed at the en-GB path can to be done with more detailed  info by accordion close to "Icons legend" within translations view, with any choosed tag.

Also and after revise, i think is better move the field "Customised en-GB files" to the tab  "en-GB development reference language" due "Customised en-GB files" needs be working with "Use development reference language" as "Yes" to have sense.

Maybe the tab, due is handling more than newest files from Github, can to be called other stuff (en-GB reference versions or Customised en-GB options or en-GB Reference and develop)

Also change "Customised en-GB files" to "en-GB - Source reference" (or source version)
and  "en-GB development reference language" to  "en-GB - Target reference" (or target version)

And explain there by "Desc" what means each stuff.

"Looks like your screenshots compare both en-GB instances and not the language we work on compared with an en-GB instance."

I don't understand that at all.

To get the diff between en-GB files we need a source en-GB reference and a target en-GB reference, and one time we have the diff between both we can make notice what is required update over any other xx-XX language tag.

Before add the "customised en-GB files" field that "source" was always our installed en-GB language files and out target reference was the files at Github in master branch (or any other allowed to choose)

After add the "customised en-GB files" field that "source" now also can to be based on other Joomla releases to choose from a list.

The no possible is use 2 source references or 2 target references at same time.

"When we compare and solve a Modified value (by changing the value or NOT, as it is often only an en-GB grammar stuff...), we should have a way to mark that string as translated (or the file complete again) in order to not have to edit again all en-GB modified values in the en-GB files to get rid of the icon"

It is not a problem add a checkbox only for modified values, but the program can not know if it is an en-GB grammar case or not.

For that cases we can edit the en-GB file and insert the new text value to solve, but with a check box we need to do the same: edit the en-GB file >> select the checkboxes with grammar cases >> Save the en-GB file. So,do that at en-GB means "grammar case".

And this one will help, with checkboxes or not, to hide that ones for any other xx-XX language tag.

But when is time to hide the real text changes at xx-XX tags, each selected checkbox there means "revised" and each translation can have "revised" the text changes or not, so also will be required edit tag by tag >> choose  cases >> and save. With or without checkboxes we need edit and save the affected files.

I really wanna mean that edit, choose modified values and save the language file are the required steps to configure and hide all "grammar cases" by en-GB tag or all "revised cases" for any xx-XX tag.

"Also, we should have a new filter in the Translations view to display only files with untranslated OR/AND modified strings in the Manager."
Yes, this one is easy to make.

Regards :)

**Comment by Valc** [comment]
No more time this weekend.

Now is working in improved mode than make it run it faster than before (one time we have the desired set of files from Github)

I have applied the commented changes in my last post.

Now all the configurable fields are together at same tab:

simple24

Now is added an accordion  to display en-GB references info (only when ref tag is en-GB) 
simple20

Added the filter:

simple21

Here the filter working:

simple23

I have founded a way to make a list of "Knowed Joomla releases" automatically, but is pending for the next weekend.

Also is pending a lot of JText to add.

And the check box for modified values.

Can you help me with the appropriate words to use with the new JText added?

Regards, Valc.

**Comment by infograf768** [comment]
Will test.
BTW, using the last iteration of the PR (therefore before your last changes), I did not get the new ini file (lib_fof) in the translations view. I had to stop using reference to development to get it.

**Comment by Valc** [comment]
I had missing add this code at line 1330 in localise.php helper.

|| !array_key_exists($file_to_include, $sha_files_list)

Thanks, the bug is solved:
libfof

it needs wait for the next selected 'interval' time to update the new files only present in develop.

**Comment by infograf768** [comment]
Issues here with Administrator Modified values. Site ones are correctly displayed.

Also, choosing Local Installed instance throws an error.
"COM_LOCALISE_ERROR_GITHUB_GETTING_LOCAL_INSTALLED_FILES"

**Comment by Valc** [comment]
I have added Jtext for that error.

What version is showing your installed Joomla's package?

Note that source reference is working fine with actual Joomla releases, but not with the not releases yet.

The normal working mode is use the last (or previous to the last ) Joomla released as source reference and then get the modified values or new keys in development using as target reference the staging or master trunk.

If previous Joomla is released and at staging or master trunk is changed the "version" to the next one, the program can not found that version of local installed files such as released due that release is not done.

Testing J 3.4.1 or prior versions as source versus last in staging or master trunk should be the normal use.

We also can test after a release with staging or master packs due the "local" language files to get from github as source reference are equal to the displayed Joomla version on that staging or master pack (normally equal to the last one released), but if there is displayed a "future version" the feature starts in "protection mode" and does not allow get files from where does not know how to get them.

So, when this error is displayed means that there are not a knowledge way to get the language files for your installed Joomla version and the feature is forced to use only that set of language files as source reference versus the staging or master trunk (your choose from the list of available packs will be ignored). And if in that case we are forced to use the language files of an staging or master pack close to the end of develop time as source ref versus the same as target ref, no great differences to display can to be founded.

If not, it is a bug and i need more info.

Regards, :)

**Comment by infograf768** [comment]
One thing at a time:

> Issues here with Administrator Modified values. Site ones are correctly displayed.

This means a bug as we do not get modifications for <b>administrator</b> part. (tested with a 3.4.1 pack)
1. In the translations view=> no icons
2. When editing a file we know has modifications, filtering by modified values is empty.

It work fine for the <b>site</b> part.

Concerning

> Also, choosing Local Installed instance throws an error.
> "COM_LOCALISE_ERROR_GITHUB_GETTING_LOCAL_INSTALLED_FILES"
> 
> What version is showing your installed Joomla's package?

I was using an already created 3.4.2 version which I wanted to work on further.

Therefore we do have an issue in that case. and the message is not very useful.
My hope is that this will change when you will be able to mark a modified string as "complete" for a specific reference version.

Suggestion concerning the filtering:
screen shot 2015-06-10 at 10 26 18

Imho the filter should not only display the files with modified values in the reference version of en-GB, but also the files with new strings. 
I would add this in the — Select State — filter as a unique choice "Modified files" and not as a new dropdown.
If "Enable development reference language" is not set in the component parameters, it would just display the files with new strings. If set, it will display both files with new strings and files with modifications in the values of the reference version of the files.

**Comment by Valc** [comment]
About the bug, can you test with a fresh Joomla 3.4.1 installation? I can't reply the issue.

If after try with new installation is working again can you use "Meld" or similar tool to take the diff between both localise folders? Comparing the bugged installation with the working installation can help a lot to understand what is happening.

For what i can to understand seems that the bugged issue is related with:

A) You have at administration language folder, language files with equal version to the last in staging or master trunk.

B) You can't get from Github the staging or master trunk.

C) Your stored copy of staging or master language files in ../media/com_localise/develop/github/joomla-cms/en-GB/administrator is empty or incomplete.

About case A), the program have not allowed to do that, but can to be possible installing master or staging packages: nothing to show as modified in that cases  (not a bug)

About case B) should be solved with right "branch" name or "token" fields configuration, or good internet connection (not a bug).

About case C)
I have pending to do add control at few code locations to verify that we are using complete sets of downloaded files from Github: The simple way is verify that the en-GB.xml file is present within that stored copies in "media" path (due following the downloads order that file is coming close to the end of the list, after type ini files, and the file is common for admin, site and installation clients.

So, if you can ensure that is not Case A or Case B, then try to verify that if in ../media/com_localise/develop/github/joomla-cms/en-GB/administrator you have the en-GB.xml file.

If no file present we have a bug saving files there and marking that task as done when not true (i have added control against it, so i don't understand how it is happening if the program will try to get the pending files to download for master or staging branches the next call).

> "Therefore we do have an issue in that case. and the message is not very useful."

Yes, the message now is not useful due i have no way to determine for sure if it is due an internet connection failure or due a "future version". One time i make the code for "Knowed releases list" than can get automatically all the Joomla releases i can to improve that message due i can to ensure that the used version does not exist as released.

> "My hope is that this will change when you will be able to mark a modified string as "complete" for a specific reference version.

That one is easy to do, i will try to make it next weekend, but this can not solve "future versions" uses. With "future versions" uses the ability to choose "customised references" is disabled due is required enable protection about that set of "unknown" language files. Note that if this feature is disabled we need be able to comeback to the installed Joomla version and the way is be able to get a copy of that release from Github.

> Imho the filter should not only display the files with modified values in the reference version of en-GB, but also the files with new strings.
> I would add this in the — Select State — filter as a unique choice "Modified files" and not as a new dropdown.
> If "Enable development reference language" is not set in the component parameters, it would just display the files with new strings. If set, it will display both files with new strings and files with modifications in the values of the reference version of the files.

For me the "Select State" is filtering by files state and we are talking about "content" files state, and that is not the same, so, i am not sure if is good share that filter for a distinct case.

For me the good idea about the explained is have a "Select content state" filter with the "Incomplete", "Complete" options, where "incomplete" means the list of files with untranslated strings or not revised modified values.

Regards

**Comment by infograf768** [comment]
> For me the good idea about the explained is have a "Select content state" filter with the "Incomplete", "Complete" options, where "incomplete" means the list of files with untranslated strings or not revised modified values.

:+1: 

Concerning my issue with administrator files, all are indeed absent from ../media/com_localise/develop/github/joomla-cms/en-GB/administrator/

I tried to delete all your .txt files (please explain their use) to solve the issue. but no luck.

**Comment by Valc** [comment]
> Concerning my issue with administrator files, all are indeed absent from ../media/com_localise/develop/github/joomla-cms/en-GB/administrator/

I have added more control about incomplete or empty sets of language  files and him messages.

If after this you continues with an empty administration folder at media "develop" folder seems a B) case.

> I tried to delete all your .txt files (please explain their use) to solve the issue. but no luck.

At the end, this files will be replaced with a DB table. For me now is easiest use that txt files to know the status of few stuffs, and for me also is less tedious to update or discard than a table.

For example, now the "client_version_customised_ref.txt" at ../administrator/components/com_localise/customised folder has been discarted due no longer used.

About how it is working now:

Others at same folder type "client_last_source_ref.txt" are used to know the last source en-GB files version moved to core language folder.

Comparing that stored version with the selected from the 'en-GB - Source files' list, if equal, is not required make a Github download or a core language files update: so, we go faster.

Other types at ../administrator/components/com_localise/develop folder are storing the SHA for each file within staging or master trunk. So, it will help to download from Github only the new files or changed SHA ID files (and also works faster).

Regards

**Comment by Valc** [comment]
I have issues with travis test. At home is installed the last version and no error messages are displayed:

> There were no issues detected.
> You are already using composer version 89540c5690c439a3c9d35f689d3d6cca8d05642c.

Any idea?

**Comment by infograf768** [comment]
There were some issues with Travis. I relaunched it and it is now OK.
Will now test last iteration of this PR.

**Comment by Valc** [comment]
Thanks JM.
Now is improved the automatic releases list.

Note that 2.5 series now are allowed but can to be enabled or disabled as a trick from localise ../models/fields/releases.php at line 77.

The filter Select content status is not working at all due i have pending the "revised" check box and modify the translations view

Regards

**Comment by infograf768** [comment]
en-gb administrator files now correctly downloaded in develop. :+1: 

Continuing my tests:
A. 
even if we do not "Enable development reference language", we get in Translations view:
screen shot 2015-06-14 at 10 16 16
when "en-GB - Source files" is set to "Local installed instance"

If disabled and set to 3.4.1, we get:
screen shot 2015-06-14 at 10 19 42

I think that, when disabled, we should have only the info:
"Target reference
No modified values or new keys in development will be displayed due this feature is disabled."
(Which should be btw:
No modified values or new keys from development branches will be displayed as "Enable development reference language" is disabled.
)
We will ask @brianteeman to look at all these strings when we get further.

One issue we have here when not using en-GB "Enable development reference language", (tested here on a staging instance) is that the PR still take into account the version in the xml, even if not updated yet. I had the issue with plg_system_redirect.ini here where the new string was translated and the replaced one deleted in my fr-FR pack but still showed as untranslated. 
The en-GB xml was still set to 3.4.1 is staging.

I made a specific PR to solve that issue ( https://github.com/joomla/joomla-cms/pull/7172 ), but it may not always be the case (or forgotten by CMS maintainers)

B. 
Adding any new string present in reference in its own [New Strings] category in the ini is not a good idea imho as we lose the original order in the reference. Please reinstate former behavior.

**Comment by Valc** [comment]
> en-gb administrator files now correctly downloaded in develop. :+1:

Great!

> A.
> even if we do not "Enable development reference language", we get in Translations view:

You are using older messages (now a good amount of Jtext has been updated)

But, ok, with that PR it is issued due i had pending to  add control there.

The stuff is than with 'Enable development reference language' disabled we have only a case when is required update the core files.

Example: we are testing 2.5.28 language version as source (or prior to installed ones) and after update the core files with that version we go to disable "Enable development reference language".

Our last saved reference will be 2.5.28 version and our local installed files should be 3.4.1 version. 

So, it is required force that update to 3.4.1 version when Enable development reference language is disabled.

The pending to do was verify if our local installed files (3.4.1) was distinct to the last one moved to core folder (2.5.28), and if not, no message will be displayed.

I am using an 2.5 series case to make notice the odd scenario that we leave there if not solved when Enable development reference language is disabled.

> About "en-GB reference version info"

Now is as you have explained, but, when we have the issue such as "2.5.28" example a system message will make notice it.

> We will ask @brianteeman to look at all these strings when we get further.

Great!

> One issue we have here when not using en-GB "Enable development reference language", (tested here on a staging instance) is that the PR still take into account the version in the xml, even if not updated yet. I had the issue with plg_system_redirect.ini here where the new string was translated and the replaced one deleted in my fr-FR pack but still showed as untranslated.
> The en-GB xml was still set to 3.4.1 is staging.

This feature does not revise at XML the language files versions. Only the installed Jooma version to know which is the "local installed instance" version number. With this one :

> $installed_version = new JVersion;
> $installed_version = $installed_version->getShortVersion();

This feature does not have allowed create, update or delete  xx-XX language files or the xx-XX content directly such as can happens with the en-GB files or en-GB content.

The only think is that when a key in "frozen" en-GB release in not present yet in master or staging trunk, as seems normal, it will be not included within a "future pack", but the "translated string" in your xx-XX language will be showed as "Extra keys" list when your language pack is created.

If installing an staging or master pack the "local" version of the pack is still set to 3.4.1, the "local installed instance" to store as local reference will be the 3.4.1, but this one is not wrong at all due in this way also is allowed know all the differences from 3.4.1 against the last one in staging or master branch (from a master or staging package installation)

> I made a specific PR to solve that issue ( joomla/joomla-cms#7172 ), but it may not always be the case (or forgotten by CMS maintainers)

Great!, but as i said before the feature does not revise that one (but is good have updated it).

> B.
> Adding any new string present in reference in its own [New Strings] category in the ini is not a good idea imho as we lose the original order in the reference. Please reinstate former behavior.

I need take a look in deep and solve if is such as you have explained.

Such as now is programmed i think that is not possible lost the order creating the language pack due each file is following the keys order such as is coming from the staging or master pack files.

When an en-GB file is saved with "Enable development reference language" enabled maybe is possible that the keys not present between both en-GB versions will be added to the saved file to keep the non present yet, but without disturb creating or translating language packs.

The stuff about your issue i think is more related when we are changing between master and staging trunks and  there is not the same keys to add or keys to delete between both, so when this one happens can make change the "future pack" to download or language files stuffs to revise or translate.

In this case, is important to be sure about which is our target branch to base our next translated language pack. "Playing wrong branches" we can lost reference keys (as is normal).

**Comment by Valc** [comment]
After revise in deep, about the save files issue i have founded a bug affecting in raw mode and creating packs. I will solve next days.
Regards,

**Comment by infograf768** [comment]
> The stuff about your issue i think is more related when we are changing between master and staging trunks and there is not the same keys to add or keys to delete between both, so when this one happens can make change the "future pack" to download or language files stuffs to revise or translate.

Nope. The new key is present in both.

**Comment by Valc** [comment]
Revised, please test again.
- The translated and saved task at xx-XX files never should be lost, but i can't reply your issue now (before than now not tested).
- The en-GB files now are not allowing "[New Strings]". Really keys to delete coming from the old set of strings not present yet within master or staging branch.
- Added more system messages
- Improve en-GB raw mode edition.

Regards

**Comment by Valc** [comment]
Hi, :)
It is not finished, but now the changes are:
- Improved working mode: now is not allowed overwrite the en-GB local installed instance with other language files versions coming from Github.
- Added check boxes to handle grammar cases at en-GB and revised changes at xx-XX translations: Grammar cases are not showed as Modified values at xx-XX translations.

Pending
- Adjust few messages and control to new working mode.
- Handle 'unchanged' new string at translations view.
- Jtext, db tables and details.

Regards

**Comment by infograf768** [comment]
@Valc 
It looks like I did not test the last instance... After adding the new table `localise_revised_value`, I am able to use the `Revised` select box when editing an ini and when filtering by `modified strings`

Once used and filtering by `Incomplete` in the Translations Manager, the updated file does not show anymore. The tip is really clear:
screen shot 2015-09-29 at 18 56 46

Therefore this looks pretty good although quite slow.
Need revising some language string values.

**Comment by Valc** [comment]
Oks, I will continue the weekend.

**Comment by infograf768** [comment]
Concerning my last small commit above to this PR, see https://github.com/joomla/joomla-cms/pull/8356
I will correct com_localise itself when releasing version 14

@Valc 
I have remarked that Complete is always ticked after editing saving the file. That is an issue as one may not correct/translate all what has to be done in that file in one go.

Also, when saving a file, I always get:
screen shot 2015-11-09 at 12 42 52

**Comment by Valc** [comment]
I think than now the "complete" issue is solved. Please, test it.
Regards and sorry for the delay, i am really bussy.

**Comment by infograf768** [comment]
Thanks, I have to find a way to test without breaking the site where I create my packs. :smile:

**Comment by infograf768** [comment]
This works well now. :smiley: 
Looking if I can create the necessary new lang strings to replace some hard coded messages.
If yes, will commit to your branch before merging and releasing new version

**Comment by Valc** [comment]
Great!
Please, i need help with the language phrases of this file:
`../component/admin/views/translations/tmpl/default_references.php`
I have not added the constants there due seems more easy to make the correction in this way and one time is ready add all the required language constants. 

I also need improve a bit a commit about 'mounting language files' (it will be done in a moment today).

Regards, Valc.

**Comment by infograf768** [comment]
Please do not touch anymore as it is pretty hard to follow what you have added or just updated vs master.

What do you mean by "mounting files"?

**Comment by Valc** [comment]
I have noticed an issue that make fails this pull if not solved, related with the way about how are readed and added the language lines coming from several paths (mounteds) to the file to save.

test server - administration - localisation manager translations 2016-03-15 07-50-36

The stuff is that the file, before edit it has no errors, but after save it, if that issue is present, the file will be saved with errors and displaying the icon 'Translation file contains errors'.

This case was happening editing my 'es-ES.lib_joomla.ini' file without errors and getting errors after save it due the way about how is mounted the file before save it.

With the other way we was inserting errors at he content file and with the new code the issue is solved making notice it.

I think that this one happens due the BOM! issue or not Linux EOL present in one of the two files used to mount the final file to save (en-GB and xx-XX language files  are normally used to mount the final xx-XX translated file).

Regards

**Comment by infograf768** [comment]
I am confused about the term "Mount".
Shall we rather use the term "Parsing" ?

**Comment by infograf768** [comment]
Hmm
I think we just should not use the term "mount", just say that the line could not be added.
I never had the issue so I have no idea how to reproduce it.
working now on strings.

**Comment by Valc** [comment]
Yes, please, replace the word "mount" by other one  more easy to understand or change the message to other explaining that "The next line  can't be added due parsing errors:  "
Regards

**Comment by infograf768** [comment]
I guess we can now merge this.

**Comment by Valc** [comment]
Great!
At the end, this one called "Simple develop" when we start... have been included more things than I could imagine ;)

Thanks for your suggestions and help :)

Regard, Valc.

**Comment by infograf768** [comment]
@valc
Can you look again at
https://github.com/joomla-projects/com_localise/pull/276

**Comment by Valc** [comment]
> @valc
> Can you look again at
> #276

Yes, revised and agree with the new way you have used to handle the issue.

Other stuff: reading another pull talking about tables with utf8mb4 support, make me notice a task pending to do at this simple develop pull:

Add the pending files type 'update sql'  for those persons that are updating the com_localise from a not included versions to update, due without add the new table '#__localise_revised_values' they will not be able to use the check boxes feature.

This is the last file added by me:
`component/admin/sql/updates/mysql/4.0.12-dev.sql`

The pending is add a file for each release after the 4.0.12-dev version.

Now that this one is merged, how is handled add the pending sql files related with updates?

Regards

**Comment by infograf768** [comment]
@Valc 
Do we need a 4.0.xx-dev.sql for each version released?

If yes, then just make a PR adding these for 4.0.13, 4.0.14 and 4.0.15 as next release will be 4.0.15-dev

**Comment by Valc** [comment]
Done at  #288
